src/Pages/ProductDetail/Components/ItemPhotoInfoSec/ItemPhotoInfoSec.js

```diff
@@ -0,0 +1,295 @@
+import React, { Component } from 'react';
+import OptionArea from './Components/OptionArea';
+import './ItemPhotoInfoSec.scss';
+
+class ItemPhotoInfoSec extends Component {
+  state = {
+    showOptions: [],
+  };
+
+  componentDidUpdate(prevProps) {
+    if (prevProps !== this.props && this.props.options.length === 0) {
+      this.setState({ showOptions: [this.props] });
+    }
+  }
+
+  increaseItem = e => {
+    const { showOptions } = this.state;
+    const findIndex = showOptions.findIndex(
+      data => data.id === Number(e.target.name)
+    );
+    let updNum = [...showOptions];
+    let foundObj = showOptions.find(data => data.id === Number(e.target.name));
+    if (foundObj.counts === foundObj.stock) {
+      alert('선택 가능한 수량을 초과했습니다');
+      return;
+    } else {
+      updNum[findIndex] = {
+        ...updNum[findIndex],
+        counts: updNum[findIndex].counts + 1,
+      };
+      this.setState({ showOptions: updNum });
+    }
+  };
+
+  decreaseItem = e => {
+    const { showOptions } = this.state;
+    const findIndex = showOptions.findIndex(
+      data => data.id === Number(e.target.name)
+    );
+    let updNum = [...showOptions];
+    Number(showOptions[findIndex].counts) <= 1
+      ? (updNum[findIndex] = {
+          ...updNum[findIndex],
+          counts: 1,
+        })
+      : (updNum[findIndex] = {
+          ...updNum[findIndex],
+          counts: updNum[findIndex].counts - 1,
+        });
+    this.setState({ showOptions: updNum });
+  };
```

Comment:
아래와 같이 더 간단히 구현할 수 있겠네요.
1. 수량이 1이하라면 수량을 감소하는 함수를 실행시킬 이유가 없습니다. 따라서 빠르게 함수를 `return` 하도록 하였습니다.
2. 결국 필요한 로직은 `updNum[findIndex].counts`의 값이 1 줄어들 뿐입니다.
3. 배열과 객체는 참조형 데이터 타입으로 const로 선언해주세요.(그래도 내부 값을 변경할 수 있답니다. 이에 대해서는 추후에 더 학습해주세요! 필요하다면 runjs에서 한번 테스트해보세요. const로 선언된 배열과 객체 내부 값은 변경 가능하지만 배열이나 객체를 통째로 할당하는 것은 안됩니다.)

```suggestion
    const { showOptions } = this.state;
    const findIndex = showOptions.findIndex(
      data => data.id === Number(e.target.name)
    );
    
    if (showOptions[findIndex].counts <= 1) return;
    const updNum = [...showOptions];
    updNum[findIndex].counts = updNum[findIndex].counts - 1;
    this.setState({ showOptions: updNum });
  };
```

Reply:
오오오 감사합니다 훨씬 간편하고 가독성도 좋네요!!! 말씀해주신 다른 부분도 수정했습니다!!